python/lsst/analysis/tools/actions/vector/selectors.py

```diff
@@ -272,3 +277,71 @@ def getInputSchema(self) -> KeyedDataSchema:
 
     def __call__(self, data: KeyedData, **kwargs) -> Vector:
         return cast(Vector, data[self.vectorKey.format(**kwargs)])
+
+
+class AndSelector(VectorAction):
+
+    selectors = ConfigurableActionStructField[VectorAction](
+        doc="Selectors for selecting rows, will be AND together",
+    )
+
+    def getInputSchema(self) -> KeyedDataSchema:
+        for action in self.selectors:
+            yield from action.getInputSchema()
+
+    def __call__(self, data: KeyedData, **kwargs) -> KeyedData:
+        mask: Optional[np.ndarray] = None
+        for selector in self.selectors:
+            subMask = selector(data, **kwargs)
+            if mask is None:
+                mask = subMask
+            else:
+                mask *= subMask  # type: ignore
+        return cast(Vector, mask)
+
+
+class ThresholdSelector(VectorAction):
+    """Return a mask corresponding to an applied threshold.
+    """
+
+    op = Field[str](doc="Operator name.")
+    threshold = Field[float](doc="Threshold to apply.")
+    columnKey = Field[str](doc="Name of column")
```

Comment:
I know this is terribly inconsistent because of changing interfaces in development, but can you change this to `VectorKey` here and anywhere else you see it (though changing in other places needs tests to make sure we dont break things). I am trying to get away from the idea that these are columns in a table. Changing it at least here will keep the name from spreading further.